src/external/smartsheet.test.ts

```diff
@@ -0,0 +1,183 @@
+                ),
+            );
+        });
+
+        it('should throw error if no newsletter subscriptions are found', async () => {
+            const mockFetchSubscription = jest.fn().mockResolvedValue({ results: [] });
+            jest.spyOn(smartsheetModule, 'fetchSubscription').mockImplementationOnce(mockFetchSubscription);
+
+            const newsletterEmail = 'test@example.com';
+            await expect(unsubscribeNewsletter(newsletterEmail)).rejects.toThrow(
+                createHttpError(StatusCodes.NOT_FOUND, `No newsletter subscription found for ${newsletterEmail}`),
+            );
+        });
+
+        it('should make DELETE requests to unsubscribe', async () => {
+            const mockFetchSubscription = jest.fn().mockResolvedValue({
+                results: [{ objectId: '123' }, { objectId: '456' }],
+            });
+            jest.spyOn(smartsheetModule, 'fetchSubscription').mockImplementationOnce(mockFetchSubscription);
+
+            const mockDeleteResponse = { status: 200 };
+            const mockFetch = (global.fetch = jest.fn().mockResolvedValue(mockDeleteResponse));
+
+            const newsletterEmail = 'test@example.com';
+            await unsubscribeNewsletter(newsletterEmail);
+
+            expect(mockFetch).toHaveBeenCalledTimes(2);
+            expect(mockFetch).toHaveBeenCalledWith(
+                `https://api.smartsheet.com/2.0/sheets/${smartsheetId}/rows?ids=123`,
+                {
+                    method: 'DELETE',
+                    headers: {
+                        Authorization: `Bearer ${smartsheetToken}`,
+                        'Content-Type': 'application/json',
+                    },
+                },
+            );
+            expect(mockFetch).toHaveBeenCalledWith(
+                `https://api.smartsheet.com/2.0/sheets/${smartsheetId}/rows?ids=456`,
+                {
+                    method: 'DELETE',
+                    headers: {
+                        Authorization: `Bearer ${smartsheetToken}`,
+                        'Content-Type': 'application/json',
+                    },
+                },
+            );
+        });
+    });
+
+    describe('fetchSubscription', () => {
+        it('should return parsed response if status is 200', async () => {
+            const mockResponse = { status: 200, json: jest.fn().mockResolvedValue({}) };
+            global.fetch = jest.fn().mockResolvedValue(mockResponse);
+            const result = await fetchSubscription(user.newsletter_email);
+            expect(result).toEqual({});
+        });
+
+        it('should throw error if status is not 200', async () => {
+            const mockResponse = { status: 500, json: jest.fn().mockResolvedValue('Something went wrong') };
+            global.fetch = jest.fn().mockResolvedValue(mockResponse);
+            await expect(fetchSubscription(user.newsletter_email)).rejects.toThrow(
+                createHttpError(500, 'Something went wrong'),
```

Comment:
so i will fail if smartsheet call fails, I thought we wanted it to be non-blocking